### PR TITLE
INS-3452: switch to go modules

### PR DESCRIPTION
**Pull request by AndreyBronin**
**- What I did**
switch to go modules
upgrade watermill to v1.0.0
upgrade minimock to v3.0.5
upgrade all mocks and fix unit tests
change contract build tmp dir to `.artifacts/tmp`
build binaries and contract plugins with -mod=vendor option

building contract unit test TestContractsBuilder_Build()

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -18,7 +18,9 @@ HEAVY_BADGER_TOOL= heavy-badger
 
 ALL_PACKAGES = ./...
 MOCKS_PACKAGE = github.com/insolar/insolar/testutils
-GOBUILD ?= go build
+GOBUILD ?= GO111MODULE=on go build -mod=vendor
```
Comment (bigbes) [inline Makefile]:
```suggestion
export GOFLAGS='-mod=vendor'
export GO111MODULE=on
export CGO_ENABLED=1
```
and i suggest to use explicit `go build`/`go test` in code.

File: ledger/light/executor/metrics_registry.go [hunk 2]
```diff
@@ -31,7 +31,7 @@ var (
 	)
 )
 
-// go:generate minimock -i github.com/insolar/insolar/ledger/light/executor.MetricsRegistry -o ./ -s _mock.go -g
+//go:generate minimock -i github.com/insolar/insolar/ledger/light/executor.MetricsRegistry -o ./ -s _mock.go -g
```
Comment (sexyhamster) [inline ledger/light/executor/metrics_registry.go]: extra space needed

Comment (AndreyBronin) [reply ledger/light/executor/metrics_registry.go]: for Goland green arrow

File: .gitignore [hunk 3]
```diff
@@ -55,3 +55,7 @@ network_cache
 
 # Local doc build directory
 /docs/build
+
+# contracts generated files
+application/contract
```
[the remark this answers is not in the document]
Comment (AndreyBronin) [reply .gitignore]: Contract proxies should be in GOPATH for go plugins compilation with -mod=vendor.

**Pull request merged**